.github/workflows/AI.yml

```diff
@@ -0,0 +1,82 @@
+name: PR Summary and Code Check
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  summarize_and_check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+        
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+
+      - name: Install Requirements
+        run: |
+          python -m pip install --upgrade pip
+          pip install requests PyYAML
+
+      - name: Run AI Summary and Checks
+        env:
+          OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+          GITHUB_TOKEN: ${{ secrets.G_TOKEN }}
+        run: |
+          #!/usr/bin/env python3
+          import os
+          import sys
+          import requests
+          import json
+
+          # Gather GitHub event details
+          event_path = os.environ.get('GITHUB_EVENT_PATH')
+          with open(event_path, 'r') as f:
+              event = json.load(f)
+
+          pr_number = event['pull_request']['number']
+          repo_full_name = event['repository']['full_name']
+          token = os.environ.get('G_TOKEN')
+          openai_key = os.environ.get('OPENAI_API_KEY')
+
+          # Get PR diff
+          headers = {
+              'Accept': 'application/vnd.github.v3.diff',
+              'Authorization': f'token {token}'
+          }
+          diff_url = event['pull_request']['url'] + "/files"
+          pr_files = requests.get(diff_url, headers={'Authorization': f'token {token}', 'Accept': 'application/vnd.github.v3+json'}).json()
```

Comment:
The response from the `requests.get` call should be checked for success (e.g., status code) before attempting to parse it as JSON to prevent runtime errors.